op.c

```diff
@@ -8018,6 +8018,15 @@ Perl_utilize(pTHX_ int aver, I32 floor, OP *version, OP *idop, OP *arg)
 
         U16 shortver = S_extract_shortver(aTHX_ use_version);
 
+        if (shortver && PL_prevailing_version) {
+            /* use VERSION while another use VERSION is in scope
+             * This should provoke at least a warning, if not an outright error
+             */
+            /* downgrading from >= 5.11 to < 5.11 is now fatal */
+            if (PL_prevailing_version >= SHORTVER(5, 11) && shortver < SHORTVER(5, 11))
+                croak("Downgrading a use VERSION declaration to below v5.11 is not permitted");
```

Comment:
Why croak() instead of Perl_croak()? (We use Perl_croak() elsewhere in this function).

Does it matter?

Reply:
`croak()` is now just a macro wrapper around `Perl_croak()`.

Until recently, we didn't have a macro wrapper for `croak()` that could be used by Perl core. It only became possible when we adopted C99 and its extra `__VA_ARGS__` macro ability. There's lots of old code that has not yet been updated, which we should go around and fix up sometime.

Reply:
Ah, croak() is Perl_croak_nocontext but it seems like it doesn't matter too much if you use one or the other? Fine

Reply:
`croak()` used to be `Perl_croak_nocontext` and still is for non-core Perl.

But when we adopted C99, we made these variadic macros safe to use properly from Perl core. For more context (pardon the pun) see 13e5ba49b2cfe0add44db552ecbebb2f785aecbc